### PR TITLE
V2 Columns Block variation #133

### LINKED ISSUE #133: FR: V2 Columns Block variation
**Agile Requirement**

As a Business Owner, I want to display an image with a card as in the following design -> https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2220-85452&mode=dev so that clients can get more information on the segment they have selected and access the trucks page of the different trucks related to the segment as well as the truck builder (if they have one) 

**Description**

Edit the existing V2 columns block to allow for a white card background, a red line above the title, a text list and the addition of two possible buttons (build and explore). 

**Acceptance Criteria**

1. The columns block can have a white card background, a red line above the title, a text list and the addition of two possible buttons (build and explore). 
2. The new additions to the existing V2 columns block follow the design->  https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2220-85452&mode=dev 
3. If the truck is available in the builder, then the "build" button is displayed, if not, only the "explore" button is displayed.

### PR DESCRIPTION
**Pull request by SantiagoHomps-NC**
Fix #133 

Test URLs:
- Before: https://develop--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/shomps/columns-variation
- After: https://133-columns-variation--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/shomps/columns-variation

### CODE REVIEW

**Review by markovukiceviccn — APPROVED** [review]

File: blocks/v2-columns/v2-columns.css [hunk 1]
```diff
@@ -99,3 +99,133 @@
...
+  padding: 40px 16px;
+  width: 100%;
+}
+
+.redesign-v2 .section--with-background.v2-columns__trucks > picture {
+  overflow: hidden;
+}
+
+.redesign-v2 .section--with-background.v2-columns__trucks > picture img {
+  height: 250%;
```
[the remark this answers is not in the document]
Comment (SantiagoHomps-NC) [reply blocks/v2-columns/v2-columns.css:119]: I know... I couldn´t figure it out myself...

File: blocks/v2-columns/v2-columns.css [hunk 2]
```diff
@@ -98,4 +98,115 @@
...
+
+.v2-columns.v2-columns--trucks .v2-columns__column p.no-header {
+  margin-top: 24px;
+}
+
+/* stylelint-disable-next-line no-descending-specificity */
+.v2-columns.v2-columns--trucks .v2-columns__btn-section a.button,
+.v2-columns.v2-columns--trucks .v2-columns__btn-section a.button:not(a.button:first-of-type) {
+  margin: 16px 0 0;
+  width: 100%;
```
[the remark this answers is not in the document]
Comment (SantiagoHomps-NC) [reply blocks/v2-columns/v2-columns.css:148]: But in this case if there are 2 its only half the width right?
Screenshot 2023-12-19 at 09 51 39

File: blocks/v2-columns/v2-columns.css [hunk 3]
```diff
@@ -99,3 +99,133 @@
...
+  padding: 0;
+  list-style-type: none;
+}
+
+.v2-columns.v2-columns--trucks .v2-columns__column li {
+  line-height: 150%;
+  margin-bottom: 8px;
+}
+
+.v2-columns.v2-columns--trucks .v2-columns__column li::before {
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.css]: I have added a generic class `li--hyphen`, this needs to be used wherever needed

File: blocks/v2-columns/v2-columns.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.css]: @cogniSyb - I have provided a aspect ratio, check if my solution direction works or should be changed

File: blocks/v2-columns/v2-columns.css [hunk 4]
```diff
@@ -99,3 +99,133 @@
...
+
+.v2-columns__trucks .v2-columns__column--with-text {
+  padding: 0;
+}
+
+.v2-columns.v2-columns--trucks .v2-columns__heading {
+  margin: 28px 0 10px;
+}
+
+.v2-columns.v2-columns--trucks .v2-columns__heading::before {
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.css]: This is done

File: blocks/v2-columns/v2-columns.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.css:147]: @cogniSyb - what's expected behaviour here?

File: blocks/v2-columns/v2-columns.css [hunk 5]
```diff
@@ -98,4 +102,93 @@
...
+    background-color: var(--c-primary-white);
+    position: absolute;
+    top: calc(200% * 0.125);
+    max-width: 485px;
+    padding: 40px;
+    margin-left: 72px;
+  }
+
+  .v2-columns.v2-columns--trucks .v2-columns__heading {
+    margin-bottom: 12px;
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.css]: In general i have used gap: 12px between elements, wherever required i have added the extra margin top/bottom.

File: blocks/v2-columns/v2-columns.js [hunk 6]
```diff
@@ -1,8 +1,15 @@
-import { createElement } from '../../scripts/common.js';
+import { variantsClassesToBEM, createElement } from '../../scripts/common.js';
 
 export default async function decorate(block) {
+  const blockParent = block.parentElement.parentElement;
   const blockName = 'v2-columns';
 
+  const variantClasses = ['trucks'];
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.js]: Done

File: blocks/v2-columns/v2-columns.js [hunk 7]
```diff
@@ -34,13 +41,35 @@ export default async function decorate(block) {
...
 
     const buttons = [...col.querySelectorAll('.button-container a')];
-    buttons.forEach((btn) => {
-      btn.classList.add('button', 'button--large', 'button--primary');
 
-      if (btn.parentElement.classList.contains('button-container')) {
-        btn.parentElement.replaceWith(btn);
+    if (isTruckVariant) {
+      if (!hasHeader) bodyElmts[0].classList.add('no-header');
+      blockParent.classList.add(`${blockName}__trucks`);
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.js]: The class is required at section level to style the background image , however added the variant class name to section

File: blocks/v2-columns/v2-columns.css [hunk 8]
```diff
@@ -98,4 +102,93 @@
...
+}
+
+.v2-columns.v2-columns--trucks .v2-columns__column p.no-header {
+  margin-top: 24px;
+}
+
+/* stylelint-disable-next-line no-descending-specificity */
+.v2-columns.v2-columns--trucks .v2-columns__btn-section a.button,
+.v2-columns.v2-columns--trucks .v2-columns__btn-section a.button:not(a.button:first-of-type) {
+  margin: 16px 0 0;
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.css]: Yep this is removed

File: blocks/v2-columns/v2-columns.css [hunk 9]
```diff
@@ -98,4 +102,93 @@
...
+    margin-bottom: 12px;
+  }
+
+  .v2-columns.v2-columns--trucks .v2-columns__column p.no-header {
+    margin-top: 0;
+  }
+
+  .v2-columns.v2-columns--trucks .v2-columns__btn-section a.button,
+  .v2-columns.v2-columns--trucks .v2-columns__btn-section a.button:not(a.button:first-of-type) {
+    margin-top: 32px;
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.css]: I have adjusted these changes made it consistent across all variants for column block

File: blocks/v2-columns/v2-columns.js [hunk 10]
```diff
@@ -34,13 +41,35 @@ export default async function decorate(block) {
...
+
+      const btnSection = createElement('div', { classes: `${blockName}__btn-section` });
```
[the remark this answers is not in the document]
Comment (Lakshmishri) [reply blocks/v2-columns/v2-columns.js:50]: I have left the button container class, the section class in required to wrap two buttons in same row.

**Pull request merged**